website/Makefile

```
@@ -1,58 +1,57 @@
+
+this_dir := $(shell dirname $(realpath $(firstword $(MAKEFILE_LIST))))
+chisel_dir := $(realpath $(this_dir)/..)
+relative_to_this_dir := $(notdir $(this_dir))
+
+mdoc_out := $(chisel_dir)/docs/generated
+
 buildDir ?= build
 subprojects = $(buildDir)/subprojects
 docs = $(buildDir)/docs
 website_docs_dest = docs/src/main/tut/chisel3/docs
 chisel3 = ../
+install_timestamp = $(this_dir)/node_modules/timestamp
 
-www-docs = \
-	$(shell find $(chisel3)/docs/ -name "*.md")
+# Everything not in the website directory
+# This is overly conservative but okay until we can do something more precise
+chisel-non-website = \
+  $(shell git ls-files $(chisel_dir) --exclude-per-directory $(relative_to_this_dir))
 
-www-src = \
-	$(website_docs_dest) \
-	$(shell find docs/src/main/resources) \
-	$(shell find docs/src/main/tut) \
-	$(chisel3)/README.md
+default: build
 
-# Get the latest version of some sequence of version strings
-# Usage: $(call getTags,$(foo))
-define latest
-$(shell echo $(1) | tr " " "\n" | sort -Vr | head -n1 | sed 's/^v//')
-endef
+# Using introduction as standin for all md files
+# Note this doesn't work when a markdown file is deleted
+$(mdoc_out)/introduction.md: $(chisel-non-website)
+	cd $(chisel_dir) && sbt docs/mdoc
 
+$(this_dir)/docs/introduction.md: $(mdoc_out)/introduction.md
+	mkdir -p $(shell dirname $@)
+	cp -R $(mdoc_out)/* $(this_dir)/docs/
```

I've had issues with using `mkdir -p` in Makefile build rules. I usually write this as an order-only prerequisite on the directory. E.g.:

```suggestion
$(this_dir)/docs/introduction.md: $(mdoc_out)/introduction.md | $(this_dir)/
	cp -R $(mdoc_out)/* $(this_dir)/docs/
	
$(this_dir)/:
	mkdir -p $@
```

You may be able to use a generic wildcard rule for the directory. I thought old macOS make has issues with this, though:

```
%/:
	mkdir -p $@
```

Ditto below.